### PR TITLE
Add documentation on `EXPLAIN` and `EXPLAIN ANALYZE`

### PR DESCRIPTION
**Pull request by devanbenz**
## Which issue does this PR close?

Closes https://github.com/apache/datafusion/issues/12088

### CODE REVIEW

File: docs/source/user-guide/explain-usage.md [hunk 1]
```diff
@@ -0,0 +1,644 @@
...
+### Data-scanning structures
+
+So the question is why we split parquet files into different ParquetExec while they are in the same partition? There are
+many reasons but two major ones are:
+
+1. Split the non-overlaps from the overlaps so we only need to apply the expensive deduplication operation on the
+   overlaps. This is the case of this example.
+2. Split the non-overlaps to increase parallel execution
+
+##### When we know there are ovelaps?
```
Comment (devanbenz) [inline docs/source/user-guide/explain-usage.md:221]: @alamb can overlaps happen in normal datafusion query plans? Or is this specific to InfluxDB? If possible could you point me in the direction of getting a parquet data-set with overlaps so I can run the EXPLAIN's locally?

Comment (devanbenz) [reply docs/source/user-guide/explain-usage.md:221]: Actually taking a look at the datafusion docs I'm not seeing `DeduplicateExec` in the public crate docs so I'm assuming this is not valid for core datafusion. 
https://docs.rs/datafusion/latest/datafusion/physical_plan/union/struct.UnionExec.html?search=DeduplicateExec

**Comment by alamb** [comment]
This is awesome -- thank you @devanbenz 

I took the liberty of pushing some commits to the branch that updated the first example and some wording -- let me know what you think. Maybe you could apply some of changes to the rest of the writeup too

I will try and find some more time later today or over the weekend to improve this documentation further with you.

**Comment by alamb** [comment]
I am going to try and document the analyze portion as well

File: docs/source/user-guide/explain-usage.md [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+
+For example, this query from ClickBench finds the top 10 users by the number of hits:
+
+```sql
+SELECT "UserID", COUNT(*) FROM 'hits.parquet' GROUP BY "UserID" ORDER BY COUNT(*) DESC LIMIT 10;
+```
+
+##### Query and query plan
+
+TODO: devanbenz can you update this section, trying to follow the model of the simpler one above?
```
Comment (alamb) [inline docs/source/user-guide/explain-usage.md:237]: @devanbenz  -- this is so cool. Thank you for getting the ball rolling. 

I wonder if you might be willing to try and take a shot at filling in this section for 

```sql
SELECT "UserID", COUNT(*) FROM 'hits.parquet' GROUP BY "UserID" ORDER BY COUNT(*) DESC LIMIT 10;
```

? 

The examples from our influx docs didn't translate very well to hits.parquet -- I apologize for not realizing that sooner

Comment (devanbenz) [reply docs/source/user-guide/explain-usage.md:237]: That sounds good to me! I'll have some time over the weekend to tackle it :)

**Comment by devanbenz** [comment]
@alamb Please let me know if theres anything else you would like for me to elaborate on within the aggregate plan. Taking a look at the original issue section I was seeing: https://github.com/apache/datafusion/issues/12088#issuecomment-2300489412 which added some insight on exchange-based parallelism. Would you like for me to elaborate more on this?

**Comment by alamb** [comment]
> @alamb Please let me know if theres anything else you would like for me to elaborate on within the aggregate plan. Taking a look at the original issue section I was seeing: [#12088 (comment)](https://github.com/apache/datafusion/issues/12088#issuecomment-2300489412) which added some insight on exchange-based parallelism. Would you like for me to elaborate more on this?

Thank you @devanbenz  -- this is great 🙏 

I pushed some more clarifications but really a wonderful job explaining  / linking.

**Review by alamb — APPROVED** [review]
Thank you @devanbenz  -- this is really great.

I pushed another commit with some additional detail but this was a great job. Super helpful to get this going. Good teamwork!  ✋ 

cc @NGA-TRAN @jstirnaman  and @2010YOUY01

**Comment by alamb** [comment]
I think this is a great start and we can continue to iterate on the documentation as follow on PRs

**Pull request merged**